src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -103,11 +144,44 @@ internal class ConventionViolator
 
             public ConventionViolator(string filePath, string extension, string expectedFile)
             {
-                Path = filePath.Replace(filePath.Split('/')[filePath.Count(p => p == '/')], string.Empty);
+#if DEBUG
+                if (filePath[filePath.Length - 1] != '/' && filePath[filePath.Length - 1] != '\\')
+                {
+                    throw new ArgumentException("Path must end with directory separator", nameof(filePath));
+                }
+
+                if (extension[0] != '.')
+                {
+                    throw new ArgumentException("Extension must include period character", nameof(extension));
+                }
+#endif
+
+                Path = filePath;
                 Extension = extension;
                 ExpectedPath = expectedFile;
             }
         }
-            
+
+        internal class ConventionViolatorComparer : IComparer<ConventionViolator>
+        {
+            internal static ConventionViolatorComparer Instance { get; } = new ConventionViolatorComparer();
+
+            public int Compare(ConventionViolator x, ConventionViolator y)
+            {
+                var result = string.Compare(x.Path, y.Path, StringComparison.Ordinal);
```

Comment:
As we are comparing paths, should these comparisons be Culture invariant?

Reply:
`StringComparison.Ordinal` makes it culture invarient, no?